contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/VoldemortBuildAndPushJob.java

```diff
@@ -316,10 +321,48 @@ else if (build && push && dataDirs.size() != 1) {
         }
         if ((!build && push) || (build && !push)) {
             log.warn("DEPRECATED : Creating one build job and separate push job is a deprecated strategy. Instead create"
-                    + "just one job with both build and push set as true and pass a list of cluster urls.");
+                     + "just one job with both build and push set as true and pass a list of cluster urls.");
         }
     }
 
+    private String getMatchingServerSupportedCompressionCodec(int nodeId) {
+        AdminClient adminclient = new AdminClient(clusterUrl.get(0),
```

Comment:
This code only checks the first cluster. This kind of makes sense given that we build only once for all clusters. That being said, it might be good to be prudent here and make this whole function loop over all the clusters? It would then return the compression type only if all clusters support it... This problem can potentially be avoided by careful operations, so if you'd rather avoid adding the extra check, perhaps it would make sense pull in an SRE to ask them if that's ok...?